pkg/apis/config/v1alpha1/resourceinterpretercustomization_types.go

```diff
@@ -23,40 +25,226 @@ type ResourceInterpreterCustomization struct {
+	// Karmada provides built-in rules for several standard Kubernetes types, see:
+	// https://karmada.io/docs/userguide/globalview/customizing-resource-interpreter/#interpretstatus
+	// If StatusReflection is set, the built-in rules will be ignored.
+	// +optional
+	StatusReflection *StatusReflection `json:"statusReflection,omitempty"`
+
+	// StatusAggregation describes the rules for Karmada to aggregate status
+	// collected from member clusters to resource template.
+	// Karmada provides built-in rules for several standard Kubernetes types, see:
+	// https://karmada.io/docs/userguide/globalview/customizing-resource-interpreter/#aggregatestatus
+	// If StatusAggregation is set, the built-in rules will be ignored.
+	// +optional
+	StatusAggregation *StatusAggregation `json:"statusAggregation,omitempty"`
+
+	// HealthInterpretation describes the health assessment rules by which Karmada
+	// can assess the health state of the resource type.
+	// +optional
+	HealthInterpretation *HealthInterpretation `json:"healthInterpretation,omitempty"`
+
+	// DependencyInterpretation describes the rules for Karmada to analyze the
+	// dependent resources.
+	// Karmada provides built-in rules for several standard Kubernetes types, see:
+	// https://karmada.io/docs/userguide/globalview/customizing-resource-interpreter/#interpretdependency
+	// If DependencyInterpretation is set, the built-in rules will be ignored.
+	// +optional
+	DependencyInterpretation *DependencyInterpretation `json:"dependencyInterpretation,omitempty"`
 }
 
 // LocalValueRetention holds the scripts for retention.
 // Now only supports Lua.
 type LocalValueRetention struct {
-	// RetentionLua holds the Lua script that is used to retain runtime values to the desired specification.
+	// LuaScript holds the Lua script that is used to retain runtime values to the desired specification.
 	// The script should implement a function as follows:
-	//     retentionLua: >
+	//     luaScript: >
 	//         function Retain(desiredObj, observedObj)
 	//             desiredObj.spec.fieldFoo = observedObj.spec.fieldFoo
 	//             return desiredObj
 	//         end
 	//
-	// RetentionLua only holds the function body part, take the `desiredObj` and `observedObj` as the function
+	// LuaScript only holds the function body part, take the `desiredObj` and `observedObj` as the function
 	// parameters or global variables, and finally returns a retained specification.
 	// +required
-	RetentionLua string `json:"retentionLua"`
+	LuaScript string `json:"luaScript"`
+}
+
+// ReplicaResourceRequirement holds the scripts for getting the desired replicas
+// as well as the resource requirement of each replica.
+type ReplicaResourceRequirement struct {
+	// LuaScript holds the Lua script that is used to discover the resource's
+	// replica as well as resource requirements
+	// The script should implement a function as follows:
+	//     luaScript: >
```

Comment:
This line is necessary? It seems redundant:
```yaml
...
  retention:
    luaScript:
      luaScript: >
       ...
```

Reply:
```yaml
  retention:
    luaScript: >
      desiredObj.spec.fieldFoo = observedObj.spec.fieldFoo
      return desiredObj
```
This is expected. 
`LuaScript only holds the function body part`

Reply:
Got it.